### ISSUE TITLE
minor doc error

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4778]
|<img alt="jjinux's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jjinux reported|
|-|-|
|Trac ID|trac#4778|
|Type|defect|
|Created|2010-12-21 19:57:07Z|

http://twistedmatrix.com/documents/current/core/howto/defer.html

```
def asynchronousIsValidUser(d, user):
    d = Deferred()
    reactor.callLater(2, d.callback, user in ["Alice", "Angus", "Agnes"])
    return d
```

I'm pretty sure that function isn't supposed to receive a d parameter.

Attachments:

* [4778.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d32/d3283df58c20ef0b724c244d001fdf465cda464e/96e63451701592e7c0dfbfcdcfa070e0dde8db18.patch) (462 bytes) - added by thijs on 2011-05-13 13:27:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4778 4778
type__defect defect
reporter__jjinux jjinux
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1292961427000000 1292961427000000
changetime__1305496288000000 1305496288000000
version__None None
owner__exarkun exarkun
cc__thijs cc__ivank
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @glyph|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|ivank set owner to @thijstriemstra|
|-|-|

I tested that 4778.patch fixes the main problem with this example, but I have these suggestions:

1. Use `defer.Deferred()` instead of `Deferred()` to match the rest of the document.

2. Add an import line `from twisted.internet import defer` to the example.

3. Add an empty news file: `twisted/topfiles/4778.misc`

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [31907]) Correct some sample code in the Deferred howto

Author: thijs, exarkun
Reviewer: ivank
Fixes: [#4778](https://github.com/twisted/twisted/issues/4778)

Remove an extraneous argument from the `asynchronousIsValidUser` example,
as well as making the code valid by supplying a necessary import of the
`defer` module.
